data-semantics.md

```diff
@@ -0,0 +1,111 @@
+
+Some tags mentioned below may contain values of significant size. Handling of such values is implementation-specific: it is the responsibility of the Tracer to honor, drop, or truncate these tags as appropriate. However, care should be exercised on the part of the instrumentor, as even the generation or passing of such values to the Tracer may create undesirable overhead for the application.
+
+It is not required that all suggested tags be used, even if one is used.
+
+### Component Identification
+
+For any span, it can be useful to specify the type of component being instrumented. This is particularly recommended for instrumentation provided for libraries or modules, where end-users may have a mix of custom and library-provided instrumentation.
+
+* `component` - string
+    - Low-cardinality identifier of the module, library, or package that is instrumented.
+    - Examples:
+        - `httplib` for instrumentation of Python builtin httplib functionality
+        - `JDBC` for instrumentation of JDBC database connectors
+        - `mongoose` for instrumentation of Ruby MongoDB client connector
+* `span.kind` - string
+    - One of `client` or `server`, indicating if this span represents a client or server
+
+### HTTP Server Tags
+
+These tags are recommended for spans marking entry into a HTTP-based service.
+
+* `http.url` - string
+    - URL of the request being handled in this segment of the trace, in [standard URI format](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier).
+    - Protocol optional
+    - Examples:
+        - `https://domain.net/path/to?resource=here`
+        - `domain.net/path/to/resource`
+        - `http://user:pass@domain.org:8888`
+* `http.method` - string
+    - HTTP method of the request being handled.
+    - Case-insensitive
+    - Examples:
+        - `GET`, `POST`, `HEAD`
+* `http.status_code` - integer
+    - HTTP status code to be returned with HTTP response.
+    - Examples:
+        - `200`, `503`
+* `span.kind` - string
+    - Value of `server` should be used to indicate that this is a server-side span (see "<a href="#component-identification">Component Identification</a>")
+
+
+### Peer Tags
+
+These tags can be provided by either client-side or server-side to describe the downstream (client case) or upstream (server case) peer being communicated with.
+
+* `peer.hostname` - string
+    - Remote hostname
+* `peer.ipv4` - string
+    - Remote IP v4 address
+* `peer.ipv6` - string
+    - Remote IP v6 address
+* `peer.port` - integer
+    - Remote port
+* `peer.service` - string
+    - Remote service name
+
+
+
+## Semantic Log Use-Cases
+
+Some events are point-in-time and do not apply to an entire span; these are reported via logged events.  For instance, an error or exception may be reported via a `log` call.
+
+
+### Exception
+
+* Event: `exception`
+* Payload - map
```

Comment:
> However, would that be implemented via a log_event overload, or via a log_error or similar?

No, I was thinking just plain old `log` method: `span.log("exception", exc)`

> If not, instrumentors are going to have to figure out what types of payloads might be valid for various tracer implementations, I guess?

If we were shooting for maps, they I would suggest changing the API to take a map rather than `interface{}` / `Object`. But it feels that people would question this choice; after all, an instance of Exception is a natural representation of the exception, why make more work for people to break it apart into a map?

We have stayed away from defining semantics of the payload so far. It's definitely something that we need to collectively address and be more prescriptive, to clarify the contract between instrumentors and implementors.  But I would suggest a different issue/PR for that. Until then, an opaque `Object` is the least restrictive form.